src/test/graphql/queries/pocketSave.integration.ts

```diff
@@ -0,0 +1,183 @@
+    await db.destroy();
+    await server.stop();
+  });
+
+  beforeAll(async () => {
+    ({ app, server, url } = await startServer(0));
+    await db('list').truncate();
+    await db('list').insert([
+      {
+        api_id: '012',
+        api_id_updated: '012',
+        favorite: 0,
+        given_url: 'http://www.ideashower.com/',
+        item_id: 55,
+        resolved_id: 55,
+        status: 0,
+        time_added: date1,
+        time_favorited: date2,
+        time_read: date3,
+        time_updated: date4,
+        title: 'the Idea Shower',
+        user_id: 1,
+      },
+      {
+        api_id_updated: 'apiid',
+        api_id: 'apiid',
+        favorite: 1,
+        given_url: 'http://irctc.co.in/',
+        item_id: 987,
+        resolved_id: 987,
+        status: 2,
+        time_added: date5,
+        time_favorited: date2,
+        time_read: date5,
+        time_updated: date5,
+        title: '',
+        user_id: 1,
+      },
+      {
+        api_id_updated: 'apiid',
+        api_id: 'apiid',
+        favorite: 1,
+        given_url: 'http://www.frameip.com/voip/',
+        item_id: 551,
+        resolved_id: 551,
+        time_added: date5,
+        time_favorited: date5,
+        time_read: date5,
+        time_updated: date5,
+        title: 'Tout sur la voip',
+        status: 1,
+        user_id: 1,
+      },
+    ]);
+  });
+
+  it('should return a pocket save with all appropriate fields', async () => {
+    const variables = {
+      userId: '1',
+      itemId: '55',
+    };
+
+    const res = await request(app).post(url).set(headers).send({
+      query: GET_POCKET_SAVE,
+      variables,
+    });
+    expect(res.body.data?._entities[0].pocketSaveById.archived).toBe(false);
+    expect(res.body.data?._entities[0].pocketSaveById.archivedAt).toBe(null);
+    expect(res.body.data?._entities[0].pocketSaveById.createdAt).toBe(
+      date1.toISOString()
+    );
+    expect(res.body.data?._entities[0].pocketSaveById.deletedAt).toBe(null);
+    expect(res.body.data?._entities[0].pocketSaveById.favorite).toBe(false);
+    expect(res.body.data?._entities[0].pocketSaveById.favoritedAt).toBe(null);
+    expect(res.body.data?._entities[0].pocketSaveById.givenUrl).toBe(
+      'http://www.ideashower.com/'
+    );
+    expect(res.body.data?._entities[0].pocketSaveById.id).toBe('55');
+    expect(res.body.data?._entities[0].pocketSaveById.status).toBe('UNREAD');
+    expect(res.body.data?._entities[0].pocketSaveById.title).toBe(
+      'the Idea Shower'
+    );
+    expect(res.body.data?._entities[0].pocketSaveById.updatedAt).toBe(
+      date4.toISOString()
+    );
+  });
+  it('should return null if no item is found for the user', async () => {
+    const variables = {
+      userId: '1',
+      itemId: '10',
+    };
+    const res = await request(app).post(url).set(headers).send({
+      query: GET_POCKET_SAVE,
+      variables,
+    });
+    expect(res.body.data?._entities[0].pocketSaveById).toBe(null);
```

Comment:
do we also want to assert on error code is NOT_FOUND?

Reply:
this is just copypasta from pre-existing structure with some other expansions, so will add this too.

Reply:
believe i added a check for error response and error code along with existing null body response.